src/test/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImplTest.java

```diff
@@ -128,4 +128,78 @@ public void testHasOpenPRWithExistingKeyAndOpenPR() {
 
         assertTrue(pullRequestStore.hasOpenPullRequests(branchName, repository));
     }
+
+    @Test
+    public void testRemovePRWithNonExistingKey() {
+        testAddPRWithNewKey(); //there is an entry in pullRequestStore with different cacheKey
+        String newKey = "different-key";
+
+        PullRequestStore oldStore = pullRequestStore;
+        BitbucketPullRequest anotherBitbucketPullRequest = setupPR(newKey, BitbucketPullState.DELETED, 1);
+        PullRequestStoreImpl.CacheKey cacheKey = new PullRequestStoreImpl.CacheKey(newKey, slug, serverId);
+
+        pullRequestStore.removePullRequest(serverId, anotherBitbucketPullRequest);
+
+        assertEquals("Store has changed;", pullRequestStore, oldStore);
+    }
+
+    @Test
+    public void testRemovePRWithExistingKeyButNonExistingPR() {
+        testAddPRWithNewKey(); //there is an entry in pullRequestStore with same cacheKey
+
+        PullRequestStore oldStore = pullRequestStore;
+        BitbucketPullRequest anotherBitbucketPullRequest = setupPR(key, BitbucketPullState.DELETED, 1);
+        PullRequestStoreImpl.CacheKey cacheKey = new PullRequestStoreImpl.CacheKey(key, slug, serverId);
+
+        pullRequestStore.removePullRequest(serverId, anotherBitbucketPullRequest);
+
+        assertEquals("Store has changed;", pullRequestStore, oldStore);
```

Comment:
this will always be true, as you pass in the same PR store (but different variable names, but they are by reference). 
We need to update this assertion to be a bit more specific.
how about asserting that
1. `anotherBitbcketPullRequest` cannot be found (returns `Optional.empty`
2. that the PR inserted with `testAddPRWithNewKey` is still present (I _think_ that's what this test is testing?)